storage-node/src/services/webApi/app.ts

```diff
@@ -251,13 +251,13 @@ async function validateUploadFileParams(req: express.Request, res: express.Respo
     throw new WebApiError(`Data object ${dataObjectId} doesn't exist in storage bag ${parsedBagId}`, 400)
   }
 
-  const isObjectPending = res.locals.acceptPendingObjectsService.pendingObjectExists(dataObjectId.toString())
-  if (isObjectPending) {
-    throw new WebApiError(`Data object ${dataObjectId} already exists`, 400)
-  }
-
   const isInStorage = getDataObjectIdFromCache(dataObjectId.toString())
   if (isInStorage) {
-    throw new WebApiError(`Data object ${dataObjectId} already exists`, 400)
+    throw new WebApiError(`Data object ${dataObjectId} already exists (in storage)`, 400)
+  }
+
+  const isObjectPending = await res.locals.acceptPendingObjectsService.pendingObjectExists(dataObjectId.toString())
+  if (isObjectPending) {
+    throw new WebApiError(`Data object ${dataObjectId} already exists (pending)`, 400)
```

Comment:
I think it makes sense to check the existence of the data object in `pending` directory first and then check in the `upload` state, _doing otherwise might incorrectly report that the data object is missing (due to race condition) but in fact, it is present_

For example based on your change:
- Say the object exists in the pending dir, so `isInStorage` would be false, but then the object is moved to the upload dir so `isObjectPending` would also be false. Which is not what we want

But if we check the object in pending dir first this case would never happen. WDYT?

Reply:
Fixed in [d06dd32](https://github.com/Joystream/joystream/pull/5054/commits/d06dd3274d90afff5ad8c862ec345b593a9ed07a)